### PR TITLE
Feature/axiscutplanes

### PR DESCRIPTION
**Pull request by martinfalk**
Fixes
* `AxisAlignedCutPlane` processor used to perform OpenGL operations in `Property::onChange()` callbacks
* version converter fix in PlottingGL

Changes proposed in this PR:
 * updated and refactored `AxisAlignedCutPlane` processor

### CODE REVIEW

**Review by petersteneteg — DISMISSED** [review]

File: modules/basegl/src/processors/axisalignedcutplane.cpp [hunk 1]
```diff
@@ -29,150 +29,109 @@
...
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/volume/volumeutils.h>
+#include <modules/opengl/rendering/meshdrawergl.h>
 
 namespace inviwo {
 
+namespace {
+
+const std::vector<OptionPropertyIntOption> channelsList = {{"channel1", "Channel 1", 0},
+                                                           {"channel2", "Channel 2", 0},
```
Comment (ResearchDaniel) [inline modules/basegl/src/processors/axisalignedcutplane.cpp]: Should not the values be 0-3, or 1-4?

**Review by ResearchDaniel — DISMISSED** [review]
Double check the channel property values

File: modules/basegl/src/processors/axisalignedcutplane.cpp (hunk 1 above)
Comment (martinfalk) [reply modules/basegl/src/processors/axisalignedcutplane.cpp]: Probably yes :) Good catch! There were three more processors affected by the same problem.

File: modules/basegl/src/processors/axisalignedcutplane.cpp [hunk 2]
```diff
@@ -29,150 +29,110 @@
...
+#include <modules/opengl/texture/textureunit.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/volume/volumeutils.h>
+#include <modules/opengl/rendering/meshdrawergl.h>
 
 namespace inviwo {
 
+namespace {
+
+const std::vector<OptionPropertyIntOption> channelsList = {{"channel1", "Channel 1", 0},
```
Comment (petersteneteg) [inline modules/basegl/src/processors/axisalignedcutplane.cpp]:
```suggestion
// in optionpproperty.h
std::vector<OptionPropertyIntOption> enumeratedOptions(std::string_view name, size_t count) {
    std::vector<OptionPropertyIntOption> res;
    for(size_t i = 0; i<count; ++i) {
        res.emplace_back(fmt::format("{}{}", toLower(name), i), fmt::format("{} {}",name, i+1, i);
    }
    return res;
}


const std::vector<OptionPropertyIntOption> channelsList = enumeratedOptions("Channel", 4);
```

Comment (martinfalk) [reply modules/basegl/src/processors/axisalignedcutplane.cpp]: Incorporated in commit 8a43a5b

File: modules/base/src/processors/layercombiner.cpp [hunk 3]
```diff
@@ -71,13 +64,17 @@ LayerCombiner::LayerCombiner()
     , outport_{"outport", "Resulting Layer with combined channels"_help}
 
     , channel_{OptionPropertyInt{"dest1", "Channel 1 Out",
-                                 "Selected channel of the first input"_help, channelsList},
+                                 "Selected channel of the first input"_help,
+                                 util::enumeratedOptions("Channel", 4)},
```
Comment (petersteneteg) [inline modules/base/src/processors/layercombiner.cpp:68]: Nice! this makes me happy :)

**Review by petersteneteg — APPROVED** [review]

**Pull request merged**